guides/no-monadfail-st-inst.md

```diff
@@ -0,0 +1,71 @@
+# What
+
+CLC has approved the [proposal to remove the `ST` instance of the `MonadFail` class 
+from `base`][proposal]. `MonadFail` consists of the single method `fail`, which is
+meant to be implemented in terms of the monad itself. In contrast, the `ST` instance
+currently (GHC 9.2, `base-4.16.0.0`) have:
+
+```haskell
+    fail s = errorWithoutStackTrace s
+```
+
+This goes against the above-mentioned idea behind the `fail` method. 
+
+[proposal]: https://github.com/haskell/core-libraries-committee/issues/33
+
+# When
+
+At the very earliest the change may appear in GHC 9.4 (~ Q3 2022), but
+given that a feature freeze for 9.4 is coming soon, it's more likely to
+happen in GHC 9.6 (~ Q2 2023).
+
+# How
+
+This is a breaking change, but of limited scope. An impact analysis showed 
+that only 11 packages of current Stackage subset require updates, and all patches
+are presented here: https://github.com/hs-monadfail-st-remove or have already been
+implemented by the maintainers.
+
+The migration strategy for this change is backward-compatible: you can migrate
+already and still retain compatibility with existing GHCs. Because of this
+CLC suggests applying patches at your earliest convenience.
+
+You are affected if you use the following inside `ST` monad.
+
+1. If you use `fail`, replace it with `error`.
+
+2. If you use a failable pattern, e.g.:
+
+```haskell
+  (x:xs) <- someFunc
+```
+
+you can replace it with a combination of `bind` and `let`, e.g.:
```

Comment:
```suggestion
you can replace it with a combination of `<-` and `let`, e.g.:
```
"bind" isn't actually an operatir, it's just how we call `>>=` and `<-` in do notation.

Reply:
Thank you ñ